examples/energy-management-app/energy-management-common/src/EnergyEvseDelegateImpl.cpp

```diff
@@ -234,6 +230,81 @@ Status EnergyEvseDelegate::StartDiagnostics()
     return Status::Success;
 }
 
+/**
+ * @brief    Called when EVSE cluster receives SetTargets command
+ */
+Status EnergyEvseDelegate::SetTargets(
+    const DataModel::DecodableList<Structs::ChargingTargetScheduleStruct::DecodableType> & chargingTargetSchedules)
+{
+    ChipLogProgress(AppServer, "EnergyEvseDelegate::SetTargets()");
+
+    EvseTargetsDelegate * targets = GetEvseTargetsDelegate();
+    VerifyOrReturnError(targets != nullptr, Status::Failure);
+
+    CHIP_ERROR err = targets->SetTargets(chargingTargetSchedules);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, StatusIB(err).mStatus);
+
+    /* The Application needs to be told that the Targets have been updated
+     * so it can potentially re-optimize the charging start time etc
+     */
+    NotifyApplicationChargingPreferencesChange();
+
+    return Status::Success;
+}
+
+Status EnergyEvseDelegate::LoadTargets()
+{
+    ChipLogProgress(AppServer, "EnergyEvseDelegate::LoadTargets()");
+
+    EvseTargetsDelegate * targets = GetEvseTargetsDelegate();
+    VerifyOrReturnError(targets != nullptr, StatusIB(CHIP_ERROR_UNINITIALIZED).mStatus);
+
+    CHIP_ERROR err = targets->LoadTargets();
+    VerifyOrReturnError(err == CHIP_NO_ERROR, StatusIB(err).mStatus);
+
+    return Status::Success;
+}
+
+Status EnergyEvseDelegate::GetTargets(DataModel::List<const Structs::ChargingTargetScheduleStruct::Type> & chargingTargetSchedules)
+{
+    ChipLogProgress(AppServer, "EnergyEvseDelegate::GetTargets()");
+
+    EvseTargetsDelegate * targets = GetEvseTargetsDelegate();
+    VerifyOrReturnError(targets != nullptr, StatusIB(CHIP_ERROR_UNINITIALIZED).mStatus);
+
+    chargingTargetSchedules = targets->GetTargets();
+
+    return Status::Success;
+}
+
+/**
+ * @brief    Called when EVSE cluster receives ClearTargets command
+ */
+Status EnergyEvseDelegate::ClearTargets()
+{
+    ChipLogProgress(AppServer, "EnergyEvseDelegate::ClearTargets()");
+
+    Status status = Status::Success;
+    CHIP_ERROR err;
+
+    EvseTargetsDelegate * targets = GetEvseTargetsDelegate();
+    VerifyOrExit(targets != nullptr, status = StatusIB(CHIP_ERROR_UNINITIALIZED).mStatus);
```

Comment:
Can we avoid a `goto exit` just for the purpose of a return? That seems to force a large scope increase on err as well for no reason and this combines early return with gotoexit. This should probably be:

```cpp
EvseTargetsDelegate * targets = GetEvseTargetsDelegate();
if (targets == nullptr)
{
    return Status::Failure; /// TODO: is this right? using a CHIP_ERROR just obscures this ...
}

CHIP_ERROR err = targets->ClearTargets();
if (err != CHIP_NO_ERROR)
{
    // Status::Failure loses the original error information, so log it before it is lost.
    ChipLogError(App, "Failed to clear Evse targets: %" CHIP_ERROR_FORMAT, err.Format());
    return Status::Failure;
}

// The Application needs to be told that the Targets have been deleted
// so it can potentially re-optimize the charging start time etc
NotifyApplicationChargingPreferencesChange();
return Status::Success;
```

```

Reply:
Code restructured as suggested. We still return a Status as this get used in `chip::app::Clusters::EnergyEvse::Instance::HandleClearTargets`